contents/blog/ab-testing-guide-for-engineers.md

```diff
@@ -0,0 +1,263 @@
+
+Again, there are many calculators online to help you determine your duration, and PostHog also includes this calculator when creating a new test.
+
+A good rule of thumb is that your test duration should be between one week and one month:
+
+- **One week** is a good minimum since users may behave differently on weekends or weekdays.
+
+- **One month** is a good maximum since otherwise you may delay shipping important product changes.
+
+## How to implement a good A/B test
+
+Once you're satisfied your test meets the criteria of a good A/B test, it's time to implement it in your code. 
+
+This is typically done using [feature flags](/blog/feature-flag-benefits-use-cases) to randomly assign users to the control or test group, and then logging when users are exposed to their variant – i.e. trigger the code path that checks the feature flag. This enables you to compare metrics in aggregate between the users in your variants.
+
+It's absolutely **essential** to only log the users in your test who would actually be impacted by your changes. Users who aren't affected by your test should be excluded. If you include unaffected users, their unchanged behavior will mix in with your results, thereby watering down the impact and the clarity of your findings.
+
+To do so, ensure that checking your feature flag and logging their exposure is the *absolute* last condition you check in your code: 
+
+```
+// ❌ Incorrect. Will include unaffected users
+function showNewChanges(user) {
+  if (posthog.getFeatureFlag('experiment-key') === 'control') {
+    return false;
+  }
+
+  if (user.hasCompletedAction) {
+    return false
+  }
+
+  // other checks
+
+  return true
+}
+```
+
+```
+// ✅ Correct. Will exclude unaffected users
+function showNewChanges(user) {
+
+  if (user.hasCompletedAction) {
+    return false
+  }
+
+  // other checks
+
+  if (posthog.getFeatureFlag('experiment-key') === 'control') {
+    return false;
+  }
+
+  return true
+}
+```
+
+## Monitoring your A/B test
+
+Once you've started your A/B test, it's a good idea to check in on it 24-48 hours after launch to ensure that everything is running correctly. Here's a list of things to check:
+
+* Check your exposure logging. For example, that the volume of users assigned to the control and test variant is what you'd expect it to be.
+
+* Check your event logging – i.e. that you're receiving all the events you expect to receive, and in the right ratios.
+
+* Monitor crashes or error rates to ensure that your test is not causing any issues. 
+
+Once you're sure that everything is running smoothly, try to avoid frequently checking the results until the test is finished. Checking too often can lead to reactive decisions based on incomplete data. It's usually best to wait until the test ends to make any decisions, as you'll have the best data at that point.
```

Comment:
```suggestion
Once you're sure that everything is running smoothly, try to avoid frequently checking the results until the test is finished. Checking too often can lead to reactive decisions based on incomplete data. It's usually best to wait until the test ends to make any decisions, as you'll have more accurate data at that point.
```